halo2/src/circuit_builder.rs

```diff
@@ -17,12 +19,10 @@ use powdr_number::{BigInt, FieldElement};
 
 use super::circuit_data::CircuitData;
 
-/// Converts an analyzed PIL, fixed and witness columns to a PlafH2Circuit and publics for the halo2 backend.
-pub(crate) fn analyzed_to_circuit<T: FieldElement>(
+pub(crate) fn analyzed_to_plaf<T: FieldElement>(
```

Comment:
This file is quite hacky right now. Before, there was only one version of this function, `analyzed_to_circuit`, which took `pil`, `fixed` and `witness` and returned the `PlafH2Circuit` needed for a Halo2 proof, and the publics. This is definitely not needed for the vkey.

Right now there are 3 versions of this function here:

- `analyzed_to_plaf`: Takes (`pil`, `fixed`) and returns `Plaf`, needed to deserialize a vkey
- `analyzed_to_circuit`: Takes (`pil`, `fixed`) and returns `PlafH2Circuit`, needed to compute a vkey
- `analyzed_to_circuit_with_witness`: Takes (`pil`, `fixed`, `witness`) and returns (`PlafH2Circuit`, `publics`), needed for the proof

I still need to de-dup and cleanup a lot of stuff here.

Reply:
Ok this is a bit cleaner now.